x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/permission/DocumentPermissionsTests.java

```diff
@@ -84,42 +95,31 @@ public void testWriteCacheKeyWillDistinguishBetweenQueriesAndLimitedByQueries()
                 new BytesArray("{\"term\":{\"q1\":\"v1\"}}"),
                 new BytesArray("{\"term\":{\"q2\":\"v2\"}}"),
                 new BytesArray("{\"term\":{\"q3\":\"v3\"}}")
-            ),
-            null
+            )
         );
         documentPermissions0.buildCacheKey(out0, BytesReference::utf8ToString);
 
         final BytesStreamOutput out1 = new BytesStreamOutput();
         final DocumentPermissions documentPermissions1 = new DocumentPermissions(
-            Set.of(new BytesArray("{\"term\":{\"q1\":\"v1\"}}"), new BytesArray("{\"term\":{\"q2\":\"v2\"}}")),
-            Set.of(new BytesArray("{\"term\":{\"q3\":\"v3\"}}"))
+            List.of(
+                Set.of(new BytesArray("{\"term\":{\"q1\":\"v1\"}}"), new BytesArray("{\"term\":{\"q2\":\"v2\"}}")),
+                Set.of(new BytesArray("{\"term\":{\"q3\":\"v3\"}}"))
+            )
         );
         documentPermissions1.buildCacheKey(out1, BytesReference::utf8ToString);
 
         final BytesStreamOutput out2 = new BytesStreamOutput();
         final DocumentPermissions documentPermissions2 = new DocumentPermissions(
-            Set.of(new BytesArray("{\"term\":{\"q1\":\"v1\"}}")),
-            Set.of(new BytesArray("{\"term\":{\"q2\":\"v2\"}}"), new BytesArray("{\"term\":{\"q3\":\"v3\"}}"))
-        );
-        documentPermissions2.buildCacheKey(out2, BytesReference::utf8ToString);
-
-        final BytesStreamOutput out3 = new BytesStreamOutput();
-        final DocumentPermissions documentPermissions3 = new DocumentPermissions(
-            null,
-            Set.of(
-                new BytesArray("{\"term\":{\"q1\":\"v1\"}}"),
-                new BytesArray("{\"term\":{\"q2\":\"v2\"}}"),
-                new BytesArray("{\"term\":{\"q3\":\"v3\"}}")
+            List.of(
+                Set.of(new BytesArray("{\"term\":{\"q1\":\"v1\"}}")),
+                Set.of(new BytesArray("{\"term\":{\"q2\":\"v2\"}}"), new BytesArray("{\"term\":{\"q3\":\"v3\"}}"))
             )
         );
-        documentPermissions3.buildCacheKey(out3, BytesReference::utf8ToString);
+        documentPermissions2.buildCacheKey(out2, BytesReference::utf8ToString);
 
         assertThat(Arrays.equals(BytesReference.toBytes(out0.bytes()), BytesReference.toBytes(out1.bytes())), is(false));
         assertThat(Arrays.equals(BytesReference.toBytes(out0.bytes()), BytesReference.toBytes(out2.bytes())), is(false));
-        assertThat(Arrays.equals(BytesReference.toBytes(out0.bytes()), BytesReference.toBytes(out3.bytes())), is(false));
```

Comment:
We used to differentiate between queries and limitedByQueries. Therefore two documentPermissions are considered different even if they have the exact same set of queries, but one has it as queries and the other has it as limitedByQueries. This difference is reflected in cacheKey computation. That is why the existing code assert non-equality between `out0.bytes()` and `out3.bytes()`. But this difference is not really necessary because intersection does not care about order or where the queries come from. Therefore the current PR removes this difference which is reflected in this test.